code/game/objects/structures/xeno.dm

```diff
@@ -718,3 +718,76 @@ TUNNEL
 			to_chat(M, "<span class='warning'>\The [src] ended unexpectedly, so we return back up.</span>")
 	else
 		to_chat(M, "<span class='warning'>Our crawling was interrupted!</span>")
+
+//Xeno Sensor
+/obj/effect/alien/resin/sensor
+	name = "sensor"
+	desc = "A strange resin structure that seems to be looking around."
+	icon_state = "sensoreye"
+	density = FALSE
+	opacity = FALSE
+	anchored = TRUE
+	max_integrity = 5
+	layer = RESIN_STRUCTURE_LAYER
+
+	var/cooldown = FALSE
+	var/number = 0
+
+/obj/effect/alien/resin/sensor/Initialize()
+	. = ..()
+	START_PROCESSING(SSobj, src)
+
+/obj/effect/alien/resin/sensor/Destroy()
+	var/area/A = get_area(src)
+	for(var/X in GLOB.alive_xeno_list)
+		to_chat(X, "<span class='xenowarning'>A sensor has been destroyed at [A].</span>")
+	. = ..()
+
+/obj/effect/alien/resin/sensor/process()
+	for(var/mob/living/carbon/H in GLOB.alive_human_list)
+		if(cooldown)
+			return
+		if(H in viewers(src))
```

Comment:
You need to rework this so you don't call viewers for every alive human

Reply:
What should I do? Should I do the for on the viewers instead and then check if they are alive?

Reply:
done

Reply:
Solved